packages/flutter/lib/src/material/material.dart

```diff
@@ -2,32 +2,94 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:math' as math;
+
 import 'package:flutter/animation.dart';
+import 'package:flutter/painting.dart';
+import 'package:flutter/rendering.dart';
 import 'package:flutter/widgets.dart';
+import 'package:vector_math/vector_math_64.dart';
 
 import 'constants.dart';
 import 'shadows.dart';
 import 'theme.dart';
 
-enum MaterialType { canvas, card, circle, button }
+enum MaterialType {
+  /// Infinite extent using default theme canvas color.
+  canvas,
+
+  /// Rounded edges, card theme color.
+  card,
+
+  /// A circle, no color by default (used for floating action buttons).
+  circle,
+
+  /// Rounded edges, no color by default (used for MaterialButton buttons).
+  button
+}
 
-const Map<MaterialType, double> _kEdges = const <MaterialType, double>{
+const Map<MaterialType, double> kMaterialEdges = const <MaterialType, double>{
   MaterialType.canvas: null,
   MaterialType.card: 2.0,
   MaterialType.circle: null,
   MaterialType.button: 2.0,
 };
 
-class Material extends StatelessComponent {
+abstract class PaintableInkFeature {
```

Comment:
I expected to add more members. I agree that this could just be Paintable now. I might even just be able to pass around the paint() methods directly where I use Paintable now, instead of having this interface.